CriteoPublisherSdk/Tests/UnitTests/CR_BidManagerFeedbackTests.m

```diff
@@ -388,6 +422,89 @@ - (void)testReadyToSendOnBidStateOnEmptyBid {
   XCTAssertEqual([self.lastSentMessages count], 1);
 }
 
+#pragma mark - Ready to send management (Live Bidding)
+
+- (void)testReadyToSendOnLiveBidValidBidReceived {
+  self.cdbResponse.cdbBids = @[ self.validBid ];
+  [self configureApiHandlerMockWithCdbRequest:self.cdbRequest
+                                  cdbResponse:self.cdbResponse
+                                        error:nil];
+
+  [self fetchLiveBidForAdUnit:self.adUnit];
+
+  XCTAssertEqual(self.feedbackFileManagingMock.readWriteDictionary.count, 0);
+  XCTAssertEqual(self.lastSentMessages.count, 1);
+}
+
+- (void)testReadyToSendOnLiveBidNoBidReceived {
+  self.cdbResponse.cdbBids = @[];
+
+  [self configureApiHandlerMockWithCdbRequest:self.cdbRequest
+                                  cdbResponse:self.cdbResponse
+                                        error:nil];
+
+  [self fetchLiveBidForAdUnit:self.adUnit];
+
+  XCTAssertEqual(self.feedbackFileManagingMock.readWriteDictionary.count, 0);
+  XCTAssertEqual(self.lastSentMessages.count, 1);
+}
+
+- (void)testReadyToSendOnLiveBidTimeoutError {
+  NSError *error = [[NSError alloc] initWithDomain:NSCocoaErrorDomain
+                                              code:NSURLErrorTimedOut
+                                          userInfo:nil];
+  [self configureApiHandlerMockWithCdbRequest:self.cdbRequest cdbResponse:nil error:error];
+
+  [self fetchLiveBidForAdUnit:self.adUnit];
+
+  XCTAssertEqual(self.feedbackFileManagingMock.readWriteDictionary.count, 0);
+  XCTAssertEqual(self.lastSentMessages.count, 1);
+}
+
+- (void)testReadyToSendOnLiveBidNetworkError {
+  NSError *error = [[NSError alloc] initWithDomain:NSCocoaErrorDomain
+                                              code:(NSURLErrorTimedOut + 1)
+                                          userInfo:nil];
+  [self configureApiHandlerMockWithCdbRequest:self.cdbRequest cdbResponse:nil error:error];
+
+  [self fetchLiveBidForAdUnit:self.adUnit];
+
+  XCTAssertEqual(self.feedbackFileManagingMock.readWriteDictionary.count, 0);
+  XCTAssertEqual(self.lastSentMessages.count, 1);
+}
+
+- (void)testReadyToSendOnLiveBidInvalidBid {
+  [self configureApiHandlerMockWithCdbRequest:self.cdbRequestForInvalidBid
+                                  cdbResponse:self.cdbResponseWithInvalidBid
+                                        error:nil];
+  [self fetchLiveBidForAdUnit:self.adUnit];
+
+  XCTAssertEqual(self.feedbackFileManagingMock.readWriteDictionary.count, 0);
+  XCTAssertEqual(self.lastSentMessages.count, 1);
+}
+
+- (void)testReadyToSendOnLiveBidConsumed {
```

Comment:
Isn't this a duplicate of `testReadyToSendOnLiveBidValidBidReceived` ?
For CSM, this sentence has more sense: on bid consumed

Reply:
Done